src/emit/expression.rs

```diff
@@ -1930,3 +1951,21 @@ fn runtime_cast<'a>(
         val
     }
 }
+
+pub(crate) fn string_to_basic_value<'a>(
+    bin: &Binary<'a>,
+    ns: &Namespace,
+    input: String,
+) -> BasicValueEnum<'a> {
+    let elem = Type::Bytes(1);
+    let size = bin.context.i32_type().const_int(input.len() as u64, false);
+
+    let elem_size = bin
+        .llvm_type(&elem, ns)
+        .size_of()
+        .unwrap()
+        .const_cast(bin.context.i32_type(), false);
+
+    let init = &Option::Some(input.as_bytes().to_vec());
+    bin.vector_new(size, elem_size, init.as_ref()).into()
```

Comment:
```suggestion
    let init = Option::Some(input.as_bytes().to_vec());
    bin.vector_new(size, elem_size, init.as_ref()).into()
```
I think the ampersand is not needed. You are already calling `as_ref` for init.

Reply:
you are right, good catch